addon/model.js

```diff
@@ -167,21 +163,45 @@ function resolveValue(key, value, modelName, store, schema, model, parentIdx) {
 
 // ie an array of nested models
 function resolvePlainArray(key, value, modelName, store, schema, model) {
```

Comment:
instead of relying on `resolvePlainArray` for this let's allow `schema.computeNestedModel` to return an array.  This makes the API simpler and more consistent with how we handle `computeAttributeReference`

So if `computeNestedModel` returns an array we create an m3 nested array and if it returns `undefined` (as well as `computeAttributeReference` we'll fall back to a plain array of *unresolved* values

what do you think?